### PR TITLE
Fix bug when auth_file does not exist

### PR DESCRIPTION
**Pull request by pjanowski**
## Summary
The recent changes to `Client.save_auth_file_to_disk()` will raise an exception if the auth_file's directory does not exist. 

 

## Changes
Create the auth_file directory if it doesn't exist.

## Importance
``` python
[32mExecuting command:[39m prefect auth login -t $PREFECT_CLOUD_KEY
[33mWARNING: You logged in with an API key using the `--token` flag which is deprecated. Please use `--key` instead.[0m
Traceback (most recent call last):
  File "/usr/local/bin/prefect", line 8, in <module>
    sys.exit(cli())
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/prefect/cli/build_register.py", line 64, in inner
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/prefect/cli/auth.py", line 148, in login
    client.save_auth_to_disk()
  File "/usr/local/lib/python3.6/site-packages/prefect/client/client.py", line 286, in save_auth_to_disk
    + toml.dumps(contents)
  File "/usr/local/lib/python3.6/pathlib.py", line 1215, in write_text
    with self.open(mode='w', encoding=encoding, errors=errors) as f:
  File "/usr/local/lib/python3.6/pathlib.py", line 1183, in open
    opener=self._opener)
  File "/usr/local/lib/python3.6/pathlib.py", line 1037, in _opener
    return self._accessor.open(self, flags, mode)
  File "/usr/local/lib/python3.6/pathlib.py", line 387, in wrapped
    return strfunc(str(pathobj), *args)
FileNotFoundError: [Errno 2] No such file or directory: '/root/.prefect/auth.toml'
```

## Checklist

This PR:

- [x] adds new tests (if appropriate)
- [x] adds a change file in the `changes/` directory (if appropriate)
- [x] updates docstrings for any new functions or function arguments, including `docs/outline.toml` for API reference docs (if appropriate)

### CODE REVIEW

File: tests/client/test_client.py [hunk 1]
```diff
@@ -106,6 +110,8 @@ def test_client_save_auth_to_disk(self):
         # Old data is unchanged
         assert data[old_key] == dict(api_key="KEY", tenant_id="ID")
 
+        prefect.context.config.home_dir
+
```
Comment (madkinsz) [inline tests/client/test_client.py:114]:
```suggestion
```

File: tests/client/test_client.py [hunk 2]
```diff
@@ -84,6 +85,9 @@ def test_client_determines_tenant_id_in_expected_order(self):
         assert client._tenant_id == "DISK_TENANT"
 
     def test_client_save_auth_to_disk(self):
+
+        Path(prefect.context.config.home_dir).rmdir()
```
Comment (madkinsz) [inline tests/client/test_client.py:89]:
```suggestion

        # Ensure saving is robust to a missing directory
        Path(prefect.context.config.home_dir).rmdir()
```

**Review by madkinsz — COMMENTED** [review]
Ah thanks! This was on my todo list and got missed. Feel free to add a changes entry or I will. It'd be great to get this in before the release tonight.

**Comment by pjanowski** [comment]
Thanks, happy to contribute. Just addded a changefile and merged your suggestions

**Review by madkinsz — DISMISSED** [review]

**Comment by madkinsz** [comment]
Looks like it might need a quick `black` run

**Review by madkinsz — APPROVED** [review]

**Pull request merged**